src/transport/AdminPairingTable.cpp

```diff
@@ -27,41 +27,68 @@
 namespace chip {
 namespace Transport {
 
-CHIP_ERROR AdminPairingInfo::StoreIntoKVS(PersistentStorageDelegate & kvs)
+PersistentStorageDelegate * gStorage = nullptr;
+AdminPairingTableDelegate * gDelegate = nullptr;
+
+CHIP_ERROR AdminPairingInfo::StoreIntoKVS()
 {
+    if (gStorage == nullptr)
+    {
+        ChipLogError(Discovery, "Server storage delegate is nill, cannot store admin in KVS");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
     char key[KeySize()];
     ReturnErrorOnFailure(GenerateKey(mAdmin, key, sizeof(key)));
 
     StorableAdminPairingInfo info;
     info.mNodeId = Encoding::LittleEndian::HostSwap64(mNodeId);
     info.mAdmin  = Encoding::LittleEndian::HostSwap16(mAdmin);
+    info.mFabricId = Encoding::LittleEndian::HostSwap64(mFabricId);
+    info.mVendorId = Encoding::LittleEndian::HostSwap16(mVendorId);
 
-    return kvs.SyncSetKeyValue(key, &info, sizeof(info));
+    gDelegate->OnAdminPersistedToStorage(mAdmin, mFabricId, mNodeId);
+    return gStorage->SyncSetKeyValue(key, &info, sizeof(info));
 }
 
-CHIP_ERROR AdminPairingInfo::FetchFromKVS(PersistentStorageDelegate & kvs)
+CHIP_ERROR AdminPairingInfo::FetchFromKVS()
 {
+    if (gStorage == nullptr)
+    {
+        ChipLogError(Discovery, "Server storage delegate is nill, cannot fetch from KVS");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
     char key[KeySize()];
     ReturnErrorOnFailure(GenerateKey(mAdmin, key, sizeof(key)));
 
     StorableAdminPairingInfo info;
 
     uint16_t size = sizeof(info);
-    ReturnErrorOnFailure(kvs.SyncGetKeyValue(key, &info, size));
+    ReturnErrorOnFailure(gStorage->SyncGetKeyValue(key, &info, size));
 
     mNodeId    = Encoding::LittleEndian::HostSwap64(info.mNodeId);
     AdminId id = Encoding::LittleEndian::HostSwap16(info.mAdmin);
+    mFabricId  = Encoding::LittleEndian::HostSwap64(info.mFabricId);
+    mVendorId  = Encoding::LittleEndian::HostSwap16(info.mVendorId);
     ReturnErrorCodeIf(mAdmin != id, CHIP_ERROR_INCORRECT_STATE);
 
+    gDelegate->OnAdminPersistedToStorage(id, mFabricId, mNodeId);
```

Comment:
This does not look like the right notification.

Reply:
fixed it